test/tests/api/v2_0/skupack_tests.py

```diff
@@ -0,0 +1,376 @@
+                result = self.__client.last_response
+                data = loads(self.__client.last_response.data)
+                assert_equal(200, result.status, message=res.reason)
+
+                #Validate that the pack content of tasks has been posted
+                LOG.info(self.__tasks.get("injectableName") + "::" + self.__packFolderId)
+                Api().workflows_get_tasks_by_name(self.__tasks.get("injectableName") + "::" + self.__packFolderId)
+                result = self.__client.last_response
+                data = loads(self.__client.last_response.data)
+                assert_equal(200, result.status, message=res.reason)
+
+                #TODO
+                #Once the 2.0 temples API route is implemented(/api/2.0/templates/library), validation of the templates
+                #should be added
+
+                # TODO
+                # Once the 2.0 profile API route is implemented (/api/2.0/templates/library), validation of the profiles
+                # should be added
+
+                """Test DELETE:api/2.0/skus/:identifier/pack"""
+                Api().skus_id_delete_pack(self.__packFolderId)
+                Api().skus_id_delete(self.__packFolderId)
+                result = self.__client.last_response
+                assert_equal(200, result.status, message=result.reason)
+
+    @test(groups=['skus_api2.tests', 'api2_put_skupack'], depends_on_groups=['api2_post_skupack'])
+    def put_skupack(self):
+        """Test PUT:api/2.0/skus/pack"""
+        for n in self.__nodes:
+            if n.get('type') == 'compute':
+                # update the sku rule above (rules[0].name.contains) with a value from the cataloged node
+                node_id = n.get('id')
+                # Post the sku pack
+                self.generateTarball("Non Quanta")
+                self.__file = {'file': open(self.__skuPackTarball, 'rb')}
+                URL = config.host + '/skus/pack'
+                res = requests.post(URL, files=self.__file)
+                assert_equal(201, res.status_code, message=res.reason)
+                self.__packFolderId = res.text.split('"')[3]
+
+
+                URL = config.host + '/skus/' + self.__packFolderId + '/pack'
+                LOG.info("The URL is : " + URL)
+                self.__file = {'file': open(self.__skuPackTarball, 'rb')}
+                res = requests.request('PUT', URL, files=self.__file)
+                LOG.info(res.status_code)
+                assert_equal(200, res.status_code, message=res.reason)
+
+                #Clean up
+                Api().skus_id_delete_pack(self.__packFolderId)
+                Api().skus_id_delete(self.__packFolderId)
+
+
+    def generateTarball(self, ruleUpdate= None):
+        tarballDirs = ["profiles", "static", "tasks", "templates", "workflows"]
+        for dir in tarballDirs:
+            os.system("mkdir -p /tmp/tarball/" + dir)
```

Comment:
You probably want to use `os.makedirs()` instead of `os.system`.   It's good practice to use native methods if they exist for portability and performance reasons.